.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,27 @@
+{
+  "image": "mcr.microsoft.com/devcontainers/go",
+  "hostRequirements": {
+    "cpus": 4,
+    "memory": "8gb",
+    "storage": "64gb"
+  },
+  "features": {
+    "ghcr.io/devcontainers/features/kubectl-helm-minikube:1": {
+      "version": "latest",
+      "helm": "latest",
+      "minikube": "latest"
+    },
+    "ghcr.io/devcontainers/features/docker-in-docker:2": {},
```

Comment:
As MacOS user, the most common and user-friendly way to install and manage Docker is to install Docker Desktop, right?

We don't have Docker Desktop (Enterprise) installed in our MacOS laptops and likely won't have in the future. Instead, we're using Podman or even Buildah.

Will it work with Podman? It seems like it's using Docker tricks which won't work with Podman.

Reply:
Based on https://github.com/devcontainers/features/tree/main/src/docker-in-docker, we can see it's using the OSS Moby engine.
For buildah/podman, worth checking https://containers.dev/features